internal/osbuild/assembler_test.go

```diff
@@ -0,0 +1,157 @@
+package osbuild
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAssembler_UnmarshalJSON(t *testing.T) {
+	tests := []struct {
+		name          string
+		assembler     Assembler
+		data          []byte
+		errorExpected bool
+	}{
+		{
+			name:          "invalid json",
+			data:          []byte(`{"name":"org.osbuild.foo","options":{"bar":null}`),
+			errorExpected: true,
+		},
+		{
+			name:          "unknown assembler",
+			data:          []byte(`{"name":"org.osbuild.foo","options":{"bar":null}}`),
+			errorExpected: true,
```

Comment:
Either I'm not seeing something or this record is the same as the one above. . It would also help if you add a short comment for invalid data entries which points out exactly where the invalid piece of data is.

Reply:
The two records are indeed distinct - the first one is for an invalid JSON (with a missing brace at the end). I added comments for the first two records to make that obvious.